cpp/daal/src/data_management/finiteness_checker.cpp

```diff
@@ -192,6 +192,30 @@ DataType sumWithAVX512(size_t n, const DataType * dataPtr)
     return sum;
 }
 
+template <typename DataType>
+DataType sumWithAVX2(size_t n, const DataType * dataPtr)
+{
+    const size_t nPerInstr = 32 / sizeof(DataType);
+    DataType sum;
+    if (sizeof(DataType) == 4)
```

Comment:
Am I missing something about how the compiler works, or why is this branching necessary??

Reply:
Now I understand